### ISSUE TITLE
[BUG] In a kotlin/js project, using PlatformBuffer.wrap and then parsing the data, the results are not as expected

### ISSUE THREAD

**Issue by D10NGYANG** [DitchOoM/buffer#56]
Here is my test code

```kotlin
import com.d10ng.common.base.toByteArrayFromHex
import com.d10ng.common.base.toHexString
import com.d10ng.common.base.toUnsignedInt
import com.d10ng.common.base.toUnsignedLong
import com.ditchoom.buffer.PlatformBuffer
import com.ditchoom.buffer.wrap
import kotlin.test.Test

class Test {

    @Test
    fun test1() {
        val str = "0001A960DBD8A500006500006400010A003132333435363738"
        val hex = str.toByteArrayFromHex()
        val buf = PlatformBuffer.wrap(hex)
        val messageId = buf.readByteArray(2).toUnsignedInt()
        val cmd = buf.readByte()
        println("messageId: $messageId, cmd: $cmd")
        val data = buf.readByteArray(buf.remaining())
        println("data=${data.toHexString()}")
        val buf1 = PlatformBuffer.wrap(data)
        println("buf1.readByteArray(4)=${buf1.readByteArray(4).toHexString()}")
        //val timestamp = buf1.readByteArray(4).toUnsignedLong() * 1000L
        //println("timestamp: $timestamp")
    }
}
```

The output log is as follows:

```text
messageId: 1, cmd: -87
data=60DBD8A500006500006400010A003132333435363738
buf1.readByteArray(4)=0001A960
```

But the expected buf1.readByteArray(4) output is 60DBD8A5

## VERSIONS

- kotlin 1.9.23
- com.ditchoom:buffer 1.3.38

**Comment by thebehera** [comment]
@D10NGYANG FYI I'm still investigating, but it appears that the issue can be simplified to:
```
val array = byteArrayOf(0, 1, 2, 3, 4, 5)
val buf = PlatformBuffer.wrap(array)
buf.readBytes(3)
val bytesRead = buf.readByteArray(buf.remaining())
val buffer2 = PlatformBuffer.wrap(bytesRead) // this is where the problem lies
assertBufferEquals(buffer2, byteArrayOf(3, 4, 5))
```


Until I get a fix out can you temporarily copy the bytes in the second `PlatformBuffer.wrap`

```
```

**Comment by thebehera** [comment]
fixed in 1.3.39
